### PR TITLE
Added example to list stream settings

### PR DESCRIPTION
**Pull request by liskaj**
there were questions how to get stream thresholds - this example shows how to read them

### CODE REVIEW

File: streams/list-stream-settings.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        if (!streamSettings) {
+            continue;
+        }
+        // STEp 5 - decode settings and print thresholds
+        const settings = Encoding.decode(streamSettings);
+
+        if (!settings) {
+            continue;
+        }
+        const settingsObj = JSON.parse(settings);
```
Comment (bakharal) [inline streams/list-stream-settings.js]: I would fold it into `Encoding.decode` and give it more descriptive name. E.g. `Encoding.decodeThresholds`

File: streams/list-stream-settings.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        console.log(`Stream: ${stream[QC.Name]}`);
+        console.log(`  Thresholds:`);
+        const keys = Object.keys(settingsObj.thresholds);
+
+        for (const key of keys) {
+            console.log(`    ${settingsObj.thresholds[key].name}`);
```
Comment (bakharal) [inline streams/list-stream-settings.js]: Maybe can pretty print entire settings here.

File: streams/list-stream-settings.js (hunk 1 above)
Comment (liskaj) [reply streams/list-stream-settings.js]: Good idea - added

File: streams/list-stream-settings.js (hunk 2 above)
Comment (liskaj) [reply streams/list-stream-settings.js]: Added

File: streams/list-stream-settings.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            }
+            if (settings.thresholds[key].upper) {
+                console.log(`      upper:`);
+                printThreshold(settings.thresholds[key].upper, 8);
+            }
+        }
+    }
+}
+
+function printThreshold(threshold, spaces) {
```
Comment (bakharal) [inline streams/list-stream-settings.js:66]: Minor, but you can probably achieve the same using native methods: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#space

**Review by bakharal — APPROVED** [review]

**Pull request merged**